relay-server/src/actors/processor.rs

```diff
@@ -2750,22 +2764,39 @@ impl EnvelopeProcessorService {
 impl Service for EnvelopeProcessorService {
     type Interface = EnvelopeProcessor;
 
-    fn spawn_handler(self, mut rx: relay_system::Receiver<Self::Interface>) {
+    fn spawn_handler(mut self, mut rx: relay_system::Receiver<Self::Interface>) {
         let thread_count = self.inner.config.cpu_concurrency();
         relay_log::info!("starting {thread_count} envelope processing workers");
 
         tokio::spawn(async move {
             let semaphore = Arc::new(Semaphore::new(thread_count));
 
-            while let (Some(message), Ok(permit)) =
-                tokio::join!(rx.recv(), semaphore.clone().acquire_owned())
-            {
-                let service = self.clone();
+            // TODO: get the global config during initialization
+            let mut config_rx = self
+                .inner
+                .global_configuration
+                .send(Subscribe)
+                .await
+                .expect(
+                    "envelope processorservice failed to subscribe to global configurationservice",
+                );
 
-                tokio::task::spawn_blocking(move || {
-                    service.handle_message(message);
-                    drop(permit);
-                });
+            loop {
+                tokio::select! {
+                   biased;
+
+                   _ = config_rx.changed() => self.global_config = config_rx.borrow().clone(),
+
+                    (Some(message), Ok(permit)) =
+                        async {tokio::join!(rx.recv(), semaphore.clone().acquire_owned())} => {
```

Comment:
this was the best way that me and iker found to put a join in a select, the problem with join in select is that select expects a future, but join evaluates to a tuple of the result of the futures